src/GraphQL/ResolveFieldContextExtensions.cs

```diff
@@ -26,7 +26,7 @@ public static object GetArgument(this IResolveFieldContext context, System.Type
 
         private static bool TryGetArgument(this IResolveFieldContext context, System.Type argumentType, string name, out object result)
         {
-            var argumentName = context.Schema?.FieldNameConverter.NameFor(name, null) ?? name;
+            var argumentName = context.Schema?.NameConverter.NameForArgument(name, context.ParentType, context.FieldDefinition) ?? name;
```

Comment:
```suggestion
        var argumentName = name;
        if (!introspection_related_argument_condition)
            argumentName = context.Schema?.NameConverter.NameForArgument(name, context.ParentType, context.FieldDefinition) ?? name;
```
Then there is no need in this warning
```
  // do not use context.GetArgument<> as it will translate the argument name via the selected schema's INameConverter
```
and you can revert introspection resolvers to use `GetArgument`. I believe that this will be more correct.

Reply:
I really don't like conditions inside `GetArgument` intended to determine if the field is an introspection type or not, as this would probably involve using `GetType()` to determine the parent field type, and then comparing the type against a predetermined list.

However, I've found another solution which both lets introspection fields use `GetArgument`, has no performance impact on other fields, does not call `GetType()`, and does not require a list of introspection types within `GetArgument`.  I simply added field extensions scoped to GraphQL.Introspection that will be called instead of the standard ones.  Now any introspection type can use the same `GetArgument` syntax just like any other field resolver, and so long as it is in the `GraphQL.Introspection` namespace, the name will not be passed through the `INameConverter`.  Please review.

Reply:
Code became more complicated. Single condition inside GetArgument would be simpler but I will review.

Reply:
If you like this idea better, I could add a dummy type (e.g. `IIntrospectionGraphType`) and then `GetArgument` could check against the dummy type for the `introspection_related_argument_condition`.  Each introspection type would need to implement this type.

Reply:
This option has a drawback. Anyone can implement this interface and "become" an introspection type. Therefore, it is better to check the type name for double underscores. See `GraphType` ctor and `NameValidator`. Also there this `SchemaPrinter.IsIntrospectionType`. You can consolidate all of this into extension method in `TypeExtensions`.

Reply:
The type need not be public.  It certainly is faster code.  And the current sample is faster yet.  But your suggestion with `StartsWith("__")` makes sense - it's just the slowest option.  Well, calling `GetType()` would be even slower.

Reply:
I updated it.  I still don't like repeated string comparisons for the introspection check for every call to GetArgument.

Reply:
> The type need not be public

Are you sure?

> It certainly is faster code. And the current sample is faster yet. But your suggestion with StartsWith("__") makes sense - it's just the slowest option. Well, calling GetType() would be even slower.

It is not a place worth to optimizing. You do not notice any performance boost in real cases. We get only code complication. If you want to make sure you can provide benchmarks.

Reply:
True; at least it isn't hard-coded to a list of types.